torch/_inductor/freezing.py

```diff
@@ -212,20 +220,20 @@ def freeze(
     constant_fold(aot_autograd_gm)
 
     fuse_conv_bn(aot_autograd_gm)
-    # now, decomp batch norm if we were unable to fuse it
-    aot_autograd_gm = decompose_unfused_batchnorms(
-        aot_autograd_gm, example_inputs, preserved_arg_indices
-    )
 
     # TODO - further restrict cse ? right now needed to dedup aliasing ops
     cse_graph = fx_graph_cse(aot_autograd_gm.graph)
     aot_autograd_gm.graph = cse_graph
     aot_autograd_gm.recompile()
     freezing_passes(aot_autograd_gm)
-
     # TODO - apply legalization in pattern matcher
     torch.fx.passes.tools_common.legalize_graph(aot_autograd_gm)
     constant_fold(aot_autograd_gm)
+    # now, decompose fusible ops if we were unable to fuse it.
+    aot_autograd_gm = decompose_unfused_fusion_ops(
+        aot_autograd_gm, example_inputs, preserved_arg_indices
+    )
```

Comment:
Any impact of moving this here?

Reply:
I move this here that the weight conversions(to_mkldnn, weight_pack) have an fx tracing issue(```NotImplementedError: Cannot access storage of OpaqueTensorImpl```), so I think it is ok to move it here to avoid meet issue.

Reply:
Could we fix the fx tracing issue instead ? 

Or have separate freezing passes that work on the undecomposed operators. We could also decompose post-freezing passes only if there is no cuda device.

`decompose_unfused_fusion_ops` should ideally still happen before the freezing passes. I think some of the difficulties you were seeing around post-decomposition would be fixed if you registered a replacement pattern like https://github.com/pytorch/pytorch/blob/main/torch/_inductor/fx_passes/fuse_attention.py#L400. Those will work even with decomposition.

Reply:
I can fix the tracing issue, but for registering a replacement pattern, it is hard to work that, for convolution, there have many constant values(stride, padding, dilation), and it seems doesn't work using a replacement pattern match.

Reply:
Now, I separate mkldnn fusion passed from freezing passe to process undecomposed operators.